### ISSUE TITLE
URI inputs should sanitize anchor links from uris

### ISSUE THREAD

**Issue by jasonbahl** [wp-graphql/wp-graphql#1712]
**Problem**

When a page has a path such as `my-page/#some-anchor-link`, many clients will pass that as the id for a `nodeByUri` query. 

While we could argue that the client should sanitize this and remove the anchor from the path prior to making a WPGraphQL request, I think it's reasonable to expect this will be sent to the server by many clients. 

**Proposal**
Wherever nodes can be fetched by URI input, WPGraphQL should take into consideration that anchor links might included in the paths and resolve to the correct node.

**Comment by justlevine** [comment]
Hey @jasonbahl can you clarify the goal of this PR?

It seems that [anchors in uris already get stripped](https://github.com/wp-graphql/wp-graphql/pull/2659/commits/9749a6eb34fa1a903389910b7368d732a9b44a0b) by `NodeResolver::parse_request()`

**Comment by jasonbahl** [comment]
There's a chance it's been resolved since opening this issue?

If it's possible to query for nodes by passing the path + anchor links (and/or query Params?) we probably just need to add some tests

**Comment by jasonbahl** [comment]
Just confirming that this does seem to work as expected. Either this was never an actual issue and I created this issue to remind myself to check it out. . .or we resolved it at some point and didn't associate the resolution with this open issue. 

In any case, things are working as I expect they should.

![CleanShot 2022-12-15 at 13 19 58](https://user-images.githubusercontent.com/1260765/207958879-98735206-183f-4c77-8423-561abe21ed36.png)
